app/pdf-viewer/buffer.py

```diff
@@ -462,73 +462,73 @@ def with_invert(self, invert, exclude_image=True):
         # if exclude image is True, will find the page all image, then get
         # each image rect. Finally, again invert all image rect.
         def fn(page, pixmap, scale):
-            pixmap.invertIRect(pixmap.irect)
+            pixmap.invert_irect(pixmap.irect)
             if not exclude_image:
                 return pixmap
 
             # exclude image only support PDF document
             imagelist = None
             try:
-                imagelist = page.getImageList(full=True)
+                imagelist = page.get_images(full=True)
             except Exception:
                 # PyMupdf 1.14 not include argument 'full'.
-                imagelist = page.getImageList()
+                imagelist = page.get_images()
 
             imagebboxlist = []
             for image in imagelist:
                 try:
-                    imagerect = page.getImageBbox(image)
-                    if imagerect.isInfinite or imagerect.isEmpty:
+                    imagerect = page.get_image_bbox(image)
+                    if imagerect.is_infinite or imagerect.is_empty:
                         continue
                     else:
                         imagebboxlist.append(imagerect)
                 except Exception:
                     pass
 
             for bbox in imagebboxlist:
-                pixmap.invertIRect(bbox * page.rotationMatrix * scale)
+                pixmap.invert_irect(bbox * page.rotation_matrix * scale)
 
         return fn
 
     def add_mark_link(self):
-        if self.page.firstLink:
-            for link in self.page.getLinks():
-                annot = self.page.addUnderlineAnnot(link["from"])
-                annot.parent = self.page # Must assign annot parent, else deleteAnnot cause parent is None problem.
+        if self.page.first_link:
+            for link in self.page.get_links():
+                annot = self.page.add_underline_annot(link["from"])
+                annot.parent = self.page # Must assign annot parent, else delete_annot cause parent is None problem.
                 self._mark_link_annot_list.append(annot)
 
     def cleanup_mark_link(self):
         if self._mark_link_annot_list:
             for annot in self._mark_link_annot_list:
-                self.page.deleteAnnot(annot)
+                self.page.delete_annot(annot)
             self._mark_link_annot_list = []
 
     def mark_search_text(self, keyword):
-        quads_list = self.page.searchFor(keyword, hit_max=999, quads=True)
+        quads_list = self.page.search_for(keyword, hit_max=999, quads=True)
         if quads_list:
             for quads in quads_list:
-                annot = self.page.addHighlightAnnot(quads)
+                annot = self.page.add_highlight_annot(quads)
```

Comment:
add_highlight_annot is not exist in my Pymupdf version.

I don't know other changed functions is exist?